shared-helpers/src/views/summaryTables.tsx

```diff
@@ -289,6 +375,169 @@ export const stackedUnitSummariesTable = (
+  let availability = null
+  if (statusElements.length > 0) {
+    if (statusElements.length >= 3) {
+      const lastElement = statusElements.pop()
+      availability = `${statusElements.join(", ")} & ${lastElement}`
+    } else {
+      availability = statusElements.join(" & ")
+    }
+  }
+
+  const text = availability ?? t("listings.unitsSummary.notAvailable")
+  const subText = hasVacantUnits
+    ? `${group.unitVacancies} ${
+        group.unitVacancies === 1 ? t("listings.vacantUnit") : t("listings.vacantUnits")
+      }`
+    : ""
+
+  return { text, subText }
+}
+
+export const getAvailabilityTextForGroup = (
+  groups: UnitGroupSummary[],
+  isComingSoon?: boolean
+): { text: string; subText: string } => {
+  // Track all statuses across groups
+  const statusSet = new Set<string>()
+  let totalVacantUnits = 0
+  let hasOpenWaitlist = false
+
+  // Collect information from all groups
+  groups.forEach((group) => {
+    if (group.openWaitlist) {
+      hasOpenWaitlist = true
+      statusSet.add(t("listings.waitlist.open"))
+    }
+
+    if (group.unitVacancies > 0) {
+      totalVacantUnits += group.unitVacancies
+    }
+  })
+
+  // Add coming soon status if needed
+  if (isComingSoon) {
+    statusSet.add(t("listings.underConstruction"))
+  }
+
+  const statusElements = Array.from(statusSet)
+
+  // Combine statuses with proper formatting
+  let availability = null
+  if (statusElements.length > 0) {
+    if (statusElements.length >= 3) {
+      const lastElement = statusElements.pop()
+      availability = `${statusElements.join(", ")} & ${lastElement}`
+    } else {
+      availability = statusElements.join(" & ")
+    }
+  }
+
+  const text = availability ?? t("listings.unitsSummary.notAvailable")
+  const subText =
+    totalVacantUnits > 0
+      ? `${totalVacantUnits} ${
+          totalVacantUnits === 1 ? t("listings.vacantUnit") : t("listings.vacantUnits")
+        }`
+      : ""
+
+  return { text, subText }
+}
+
+export const stackedUnitGroupsSummariesTable = (
+  summaries: UnitGroupSummary[],
+  isComingSoon?: boolean
+): Record<string, StackedTableRow>[] => {
+  const ranges = mergeGroupSummaryRows(summaries)
+
+  const getUnitText = (ranges: StackedGroupSummary) => {
+    if (ranges.minUnitType === ranges.maxUnitType)
+      return t(`listings.unitTypes.${ranges.minUnitName}`)
+    return `${t(`listings.unitTypes.${ranges.minUnitName}`)} - ${t(
+      `listings.unitTypes.${ranges.maxUnitName}`
+    )}`
+  }
+  const getRentText = (ranges: StackedGroupSummary) => {
+    const hasPercentUnits = ranges.minPercentageRent !== null && ranges.maxPercentageRent !== null
+    const hasCurrencyUnits = ranges.minDollarRent !== null && ranges.maxDollarRent !== null
+
+    // If a listing has mixed rent type units, show more generic information
+    if (hasPercentUnits && hasCurrencyUnits) {
+      return `${t("t.ofIncome")}, ${t("t.orUpTo")} $${ranges.maxDollarRent?.toLocaleString()}`
+    }
```

Comment:
Double check if that is correct, if we have both min and max percentage rent it should display `% of income, or up to ...`